desc/plotting.py

```diff
@@ -626,8 +630,28 @@ def plot_2d(
     plot_axes = _get_plot_axes(grid)
     if len(plot_axes) != 2:
         return ValueError(colored("Grid must be 2D", "red"))
-
-    data, label = _compute(eq, name, grid, kwargs.pop("component", None))
+    component = kwargs.pop("component", None)
+    if name != "Bn":
+        data, label = _compute(
+            eq,
+            name,
+            grid,
+            component=component,
+        )
+    else:
+        field = kwargs.pop("field", None)
+        errorif(
+            field is None,
+            ValueError,
+            "If Bn is entered as the variable to plot, a magnetic field"
+            " must be provided.",
+        )
+        field_grid = kwargs.pop("field_grid", None)
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            data, _ = field.compute_Bnormal(eq, eval_grid=grid, source_grid=field_grid)
```

Comment:
will this only be correct for vacuum equilibria? What if is a nonozero Bn from the plasma field?

Reply:
Right now have it using the same plot grid for eval and source, I figure most times it will be fine. Can change it to be separate grids if needed

Reply:
I think same grid is probably fine, the plotting resolution is already usually much higher than what we use during optimization.